mlkem/native/x86_64/src/basemul.c

```diff
@@ -8,15 +8,56 @@
 #if defined(MLKEM_NATIVE_ARITH_BACKEND_X86_64_DEFAULT)
 
 #include "consts.h"
+
 #include "poly.h"
 #include "polyvec.h"
 
 #include "arith_native_x86_64.h"
 
-static void poly_basemul_montgomery_avx2(poly *r, const poly *a, const poly *b)
+static const int16_t zetas_avx2[64] = {
+#include "x86_64_zetas_mulcache.i"
+};
+
+#define QINV (-3327) /* q^-1 mod 2^16 */
+
+void poly_mulcache_compute_avx2(poly_mulcache *x, const poly *y)
+{
+  int j;
```

Comment:
Since we aim to use assembly-level verification using HOL-Light, could we implement this in ASM rather than intrinsics, please? ISTM that this should essentilly be the code following`TODO: This could be precomputed in the mulcache` in (the old) `basemul.S`.

Reply:
sure, you want me to do it as part of this PR?

Reply:
Preferably, yes.

Reply:
done.